### DISCUSSION TITLE
[Question] -  FormsFlow -> Admin -> Internal Server Error

### DISCUSSION THREAD

**Discussion by dinbtechit** [AOT-Technologies/forms-flow-ai#993] [Installation]
I have deployed formflow-ai using the Helm charts. The apps the be working very well except the "admin" tab shows an `Intenal Server Error`. Any idea what is going on?

![image](https://user-images.githubusercontent.com/17984781/199546745-deee733e-08d3-458f-bd17-11ed9cd7ce8d.png)

**Comment by sumesh-aot** [comment]
@dinbtechit Have you updated the redash key in secret? For now, admin have only redash dashboard authorization, so the API tries to call redash to get available dashboards.
redash key is dynamic can can be generated only when you setup your profile in redash. Follow the below steps;
1 - Login to redash by following the default setup.
2 - Go to Settings -> Account
3 - Copy the API Key and update Secrets -> forms-flow-ai -> INSIGHT_API_KEY in k8s with API key
4 - restart web api

**Reply by dinbtechit** [reply]
Oh ok let me give that a shot. Thank you!

**Reply by dinbtechit** [reply]
After updating the secrets and restarting the web API pod. Seeing this cors error. Any idea how to resolve this?

![image](https://user-images.githubusercontent.com/17984781/199575130-c535438e-2d19-48f6-bc46-962be73494e3.png)

Loading page:

![image](https://user-images.githubusercontent.com/17984781/199575189-fc5c8927-90ae-4842-b767-8cbd7f83f3c7.png)

**Reply by sumesh-aot** [reply]
@dinbtechit Weird! Can you attach the api logs and your jwt (which you can get from localStorage -> authToken).

**Reply by dinbtechit** [reply]
Logs: 

<details>
 <summary>Logs</summary>

 ```
 TypeError: 'NoneType' object is not iterable
 [32;1m2022-11-02 18:44:54,578 - app - INFO - Establishing new connection to formio... (startup.py:29)[0m
 [32;1m2022-11-02 18:44:54,578 - app - INFO - Role id fetching started... (formio.py:72)[0m
 [31;21m2022-11-02 18:44:54,611 - app - ERROR - Failed to fetch role ids !!! (formio.py:78)[0m
 [31;21m2022-11-02 18:44:54,611 - app - ERROR - Expecting value: line 1 column 1 (char 0) (startup.py:43)[0m
 [32;1m2022-11-02 18:44:54,612 - app - INFO - API endpoint: FormioResource.get took 0.03418850898742676 seconds 
 (profiler.py:19)[0m
 [31;21m2022-11-02 18:44:54,612 - app - ERROR - Exception on /formio/roles [GET] (app.py:1449)[0m
 Traceback (most recent call last):
   File "/venv/lib/python3.9/site-packages/flask/app.py", line 2073, in wsgi_app
     response = self.full_dispatch_request()
   File "/venv/lib/python3.9/site-packages/flask/app.py", line 1520, in full_dispatch_request
     return self.finalize_request(rv)
   File "/venv/lib/python3.9/site-packages/flask/app.py", line 1541, in finalize_request
     response = self.process_response(response)
   File "/venv/lib/python3.9/site-packages/flask/app.py", line 1880, in process_response
     response = self.ensure_sync(func)(response)
   File "/forms-flow-api/app/src/formsflow_api/resources/formio.py", line 59, in add_jwt_token_as_header
     filter(
 ```
</details>

Is there any specific information you are looking for jwt?  There is a bit more than I want to share.

**Reply by sumesh-aot** [reply]
From logs; it looks like the web-api is unable to call form.io api to collect the role ids. Can you check if form.io is up and running. Also try to login to form.io URL using the admin email and password.

**Reply by dinbtechit** [reply]
I see form.io is up and running. 

![image](https://user-images.githubusercontent.com/17984781/199581047-6d48e15d-c577-4ba3-93fb-6c91e5d822bb.png)

**Comment by SawantDarshan** [comment]
@sumesh-aot I am having same issue even after updating INSIGHT_API_KEY.
seems like api key to http://localhost:5000/formio/roles is failing 
previously we used to run script to create roles and add it as env variable.
If there's any way to fix this please let us know 
![error](https://user-images.githubusercontent.com/86104901/203803627-766ef4c1-d27a-4711-b957-be77ee2e3867.PNG)

error --->
forms-flow-webapi_1  | 2022-11-24 13:58:08,636 - app - ERROR - Failed to fetch role ids !!! (formio.py:78)
forms-flow-webapi_1  | 2022-11-24 13:58:08,637 - app - ERROR - Expecting value: line 1 column 1 (char 0) (startup.py:43)

**Reply by sumesh-aot** [reply]
@SawantDarshan Issue above was a different issue than you are seeing, as it was caused due to duplicate resources on formio.
Can you please check whether you have the FORMIO related env vars (https://github.com/AOT-Technologies/forms-flow-ai-deployment/blob/main/docker-compose/docker-compose.yml#L155-L161) in your API environment ? If not, update the values and restart API.
Also make sure, FORMIO_JWT_SECRET is same in both API config and formio config.
We moved all the formio roles and jwt to backend API to enhance security.
